test/cmdlineTests.sh

```diff
@@ -41,31 +43,33 @@ test "${output//[[:blank:]]/}" = "3"
 
 function compileFull()
 {
-    files="$*"
+    local files="$*"
+
     set +e
-    "$SOLC" --optimize \
-    --combined-json abi,asm,ast,bin,bin-runtime,clone-bin,compact-format,devdoc,hashes,interface,metadata,opcodes,srcmap,srcmap-runtime,userdoc \
-    $files >/dev/null 2>&1
-    failed=$?
+    local output=$( ("$SOLC" $FULLARGS $files) 2>&1 )
+    local failed=$?
     set -e
+
     if [ $failed -ne 0 ]
     then
         echo "Compilation failed on:"
-        cat $files
```

Comment:
Right, the problem is that those files all enter as parameters to a single execution of `solc` so you can't distinguish which one actually caused the error. In the case I was debugging, it was incorrectly checking a restriction and the error output from `solc` came out fine, indicating the file and line that triggered the false positive.

On the other hand, if it fails with an exception, I'd argue that we should make `solc` itself print the offending input file before exiting. The remaining cases (like aborts) should be very rare in integration tests.